platform/macos/app/MGLVectorSource+MBXAdditions.m

```diff
@@ -7,19 +7,23 @@ @implementation MGLVectorSource (MBXAdditions)
     static dispatch_once_t onceToken;
     static NS_SET_OF(NSString *) *mapboxStreetsLanguages;
     dispatch_once(&onceToken, ^{
-        mapboxStreetsLanguages = [NSSet setWithObjects:@"en", @"es", @"fr", @"de", @"ru", @"zh", nil];
+        // https://www.mapbox.com/vector-tiles/mapbox-streets-v7/#overview
+        mapboxStreetsLanguages = [NSSet setWithObjects:@"ar", @"de", @"en", @"es", @"fr", @"pt", @"ru", @"zh", @"zh-Hans", nil];
     });
     return mapboxStreetsLanguages;
 }
 
-+ (nullable NSString *)preferredMapboxStreetsLanguage {
-    for (NSString *language in [NSLocale preferredLanguages]) {
-        NSString *languageCode = [[NSLocale localeWithLocaleIdentifier:language] objectForKey:NSLocaleLanguageCode];
-        if ([[MGLVectorSource mapboxStreetsLanguages] containsObject:languageCode]) {
-            return languageCode;
++ (NSString *)preferredMapboxStreetsLanguage {
+    NSArray<NSString *> *supportedLanguages = [MGLVectorSource mapboxStreetsLanguages].allObjects;
+    NSArray<NSString *> *preferredLanguages = [NSBundle preferredLocalizationsFromArray:supportedLanguages
+                                                                         forPreferences:[NSLocale preferredLanguages]];
+    NSString *mostSpecificLanguage;
+    for (NSString *language in preferredLanguages) {
+        if (language.length > mostSpecificLanguage.length) {
```

Comment:
This is sort of a hack to prioritize `zh-Hans` in the case where the preference is `zh-*`. Confusingly, the [`+[NSBundle preferredLocalizationsFromArray:forPreferences:]`](https://developer.apple.com/reference/foundation/nsbundle/1409418-preferredlocalizationsfromarray) documentation claims that the method in this case returns:

> typically either a single non-region-specific localization or a region-specific localization followed by a corresponding non-region-specific localization as a fallback

whereas `preferredLanguages` actually ends up with the value `["zh", "zh-Hans"]`.

Unfortunately, it is the case that, when the preference is `zh-Hant` or `zh-Hant-*`, `preferredLanguages` ends up being `["en"]`. If we do add a `zh-Hant`, `zh-Hant-TW` will correctly match `zh-Hant`, however.